app/main/search_service.py

```diff
@@ -0,0 +1,67 @@
+import os
+from elasticsearch import Elasticsearch
+
+# Setup
+es_url = os.getenv('DM_ELASTICSEARCH_URL')
+es = Elasticsearch(es_url)
+
+
+# Basic keyword query without filters endpoint - may not be required as the
+# filtered query gives the same results if only a keyword query is supplied
+#
+# I'm leaving it here for now though
+#
+# def keyword_query(query, start_from=0, size=10):
+#     res = es.search(index="services",
+#                     body={
+#                         "from": start_from, "size": size,
+#                         "fields": ["id", "serviceName", "lot",
+#                                    "serviceSummary"],
+#                         "query": {
+#                             "multi_match": {
+#                                 "query": query,
+#                                 "fields": ["serviceName",
+#                                            "serviceSummary",
+#                                            "serviceFeatures",
+#                                            "serviceBenefits"],
+#                                 "operator": "or"
+#                             }
+#                         }
+#                     }
+#                     )
+#     return res
+
+
+def keyword_query_with_filters(request_args, start_from=0, size=10):
+    query = request_args["q"]
+    filters = []
+    for k, v in request_args.iteritems():
+        # Do not include the query string as a filter
+        if k == "q":
+            continue
+        filters.append({"term": {k: v.lower()}})
```

Comment:
I'd suggest considering namespacing the filter parameters here.  If you want to extend this API in future (eg, to add facetting options, or similar), you'd risk the new arguments colliding with existing filter ones.

For the GOV.UK search API, we prefix filter fields with `filter_` - eg, `filter_organisations=hmrc`.

I think there might be some useful other lessons like this you could take from the GOV.UK search API - both in terms of things we've done, and things we might have been better off not doing.  I'd be happy to have a chat sometime next week about this.